main/src/main/java/org/opennms/nephron/RefType.java

```diff
@@ -31,189 +31,347 @@
 import java.io.IOException;
 import java.io.InputStream;
 import java.io.OutputStream;
+import java.util.Objects;
 
 import org.apache.beam.sdk.coders.Coder;
 import org.apache.beam.sdk.coders.NullableCoder;
 import org.apache.beam.sdk.coders.StringUtf8Coder;
 import org.apache.beam.sdk.coders.VarIntCoder;
 import org.opennms.nephron.elastic.ExporterNode;
 import org.opennms.nephron.elastic.FlowSummary;
+import org.opennms.netmgt.flows.persistence.model.Direction;
 import org.opennms.netmgt.flows.persistence.model.FlowDocument;
+import org.opennms.netmgt.flows.persistence.model.NodeInfo;
+
+import com.google.common.base.Strings;
 
 /**
- * Describes a dimensions flow data can be grouped into.
+ * Describes the dimensions flow data can be grouped into.
  */
-public abstract class RefType<T extends Ref> {
+abstract class RefType {
```

Comment:
```suggestion
public abstract class RefType {
```

Any reason why this should not be public?